packages/comms/src/ecl/xsdParser.ts

```diff
@@ -88,31 +88,31 @@ export class XSDXMLNode extends XSDNode {
                 break;
             default:
                 const numStr: string = "0123456789";
-                const underbarPos: number = this.type.lastIndexOf("_");
-                const length: number = underbarPos > 0 ? underbarPos : this.type.length;
+                const underbarPos: number = this.type?.lastIndexOf("_") ?? 0;
```

Comment:
seems like it should be ` ?? -1` instead. That's what it would've returned if the "_" wasn't found, not that it's the first character

Reply:
Updated...